feat: searxng setup

nabeelraza-7 opened the pull request:
Ran using `docker compose up -d`

saad039 commented on docker-compose.yaml:
```diff
@@ -12,42 +12,34 @@ services:
+    ports:
+      - "8081:8080"
+    volumes:
+      - ./searxng:/etc/searxng:rw
```
I'd recommend you to use a named-volume instead of a local volume here.

nabeelraza-7 replied:
I'm actually mapping the folder `searxng`, with some config files, from the codebase to the `/etc/searxng`. I don't know what the best practice that should be implemented here but the cursor also recommended this approach.

![image](https://github.com/user-attachments/assets/c0faf830-03a9-495e-9cea-acf4fc2cdbb3)

Do let me know if there's still something I can do.

nabeelraza-7 replied:
https://gpt.antematter.io/s/07da5ed4-fea8-4ce9-aaa8-11f8a12aeb6c

nabeelraza-7 replied:
Made some changes, please take a look

paddy313 committed:
Add Mistral OCR integration and configuration support

tjbck committed:
refac

tjbck committed:
Merge pull request #12168 from yak1ex/fix_inconsistent_value_check_in_onMessageHandler_in_Chat_svelte

fix: Inconsistent value check in onMessageHandler() in Chat.svelte

JunaidPinjari committed:
Fix: CSV loader encoding issue using autodetect_encoding=True

JunaidPinjari committed:
Feat: Add optional ingest_file and as_attachment params to upload and download endpoints

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
Merge pull request #12176 from JunaidPinjari/feat/file-upload-download-options

Support skipping ingestion and downloading files as attachments

tjbck committed:
refac

tjbck committed:
Merge pull request #12170 from JunaidPinjari/fix/csv-autodetect-encoding

Fix: CSV loader encoding issue using autodetect_encoding=True

tjbck committed:
fix: ollama requests

tjbck committed:
refac: html rendering

tjbck committed:
feat: goto message

tjbck committed:
chore: format

tjbck committed:
refac

tjbck committed:
refac: styling

tjbck committed:
enh: tool args parse json fallback

rgaricano committed:
Fix alignament buttons text

Fix alignament buttons text

Fixed align tag on user & admin settings

- correction of some align tag assigned to buttons that are right and
have to be left, this means that although it is not noticeable in
English, in other languages, especially if several lines appear, it is
obvious.

Ithanil committed:
adjust jpeg compression quality to 0.15 for avatar uploads

TiancongLx committed:
i18n: update zh-TW

tjbck committed:
Merge pull request #12194 from TiancongLx/dev

i18n: update zh-TW

tjbck committed:
Merge pull request #12192 from Ithanil/avatar_compression

feat: adjust jpeg compression quality to 0.15 for avatar uploads

tjbck committed:
revert

tjbck committed:
Merge pull request #12191 from rgaricano/dev

Fix alignament buttons text

tjbck committed:
Merge pull request #12132 from Phlogi/dev-fetch-documents-once

Avoid multiple data fetching

tjbck committed:
fix/refac: hybrid search

tjbck committed:
feat: prompt autocompletion user setting

tjbck committed:
refac: rm profile image from feedback user object

tjbck committed:
feat: user webhooks system settings

tjbck committed:
Merge pull request #11307 from wangtunan/fix/stop_task_func_need_use_async_await_syntax

fix: stopTask func need use async/await syntax

tjbck committed:
Merge pull request #8594 from jayteaftw/main

feat: Support for instruct/prefixing embeddings

tjbck committed:
refac: embedding prefix var naming

tjbck committed:
refac

tjbck committed:
refac: styling

tjbck committed:
refac: folders

aleixdorca committed:
Update catalan translation.json

tjbck committed:
Merge pull request #12218 from aleixdorca/dev

Update catalan translation.json

tjbck committed:
fix: model import/export

tjbck committed:
refac: prompt command behaviour

tjbck committed:
refac: floating buttons behaviour in shared chat

tjbck committed:
Merge pull request #11087 from tarmst/fix-files-access-control

fix: Add access control usage to files APIs

tjbck committed:
refac: knowledge file ac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac: styling

Phlogi committed:
Fix middleware.py missing prefix on embedding function

rgaricano committed:
FIX: Translation (es-ES) - Spanish

Complete Revision of Spanish Translation (es-ES)

Revised spanish translation.
Standardization of terminology, vocabulary, and syntax.
Revision of incorrectly explained or defined terms.

Phlogi committed:
Run hybrid_search in parallel

rgaricano committed:
FIX: Translation (es-ES) - Spanish rev.

Complete Revision of Spanish Translation (es-ES)

Revised spanish translation.
Standardization of terminology, vocabulary, and syntax.
Revision of incorrectly explained or defined terms.

(added two lost recently updated entries)

rgaricano committed:
FIX: Translation (es-ES) - Spanish

Complete Revision of Spanish Translation (es-ES)

Revised spanish translation.
Standardization of terminology, vocabulary, and syntax.
Revision of incorrectly explained or defined terms.

rgaricano committed:
Update translation.json

fix X typo
lines 4 & 304

tjbck committed:
refac: rm log info

tjbck committed:
Merge pull request #12238 from rgaricano/dev

PR: Translation (es-ES) - Spanish v0.6.0

tjbck committed:
chore: format

tjbck committed:
Merge pull request #12236 from Phlogi/fix_prefix

Fix middleware.py missing prefix on embedding function

tjbck committed:
refac

tjbck committed:
feat: hide base model

tjbck committed:
Merge pull request #12239 from Phlogi/dev-threads-on-hybrid

perf: parallelize hybrid search

tjbck committed:
feat: public sharing permissions

tjbck committed:
feat: public sharing permissions

Co-Authored-By: Taylor Wilsdon <6508528+taylorwilsdon@users.noreply.github.com>

tjbck committed:
feat: enforced temporary chat

tjbck committed:
refac: styling

tjbck committed:
chore: format

tjbck committed:
doc: changelog

tjbck committed:
doc: wording

tjbck committed:
doc: wording

tjbck committed:
doc: changelog

tjbck committed:
doc: wording

tjbck committed:
refac

tjbck committed:
Merge pull request #11728 from open-webui/dev

0.6

dependabot[bot] committed:
build(deps): bump python-multipart from 0.0.18 to 0.0.20 in /backend

Bumps [python-multipart](https://github.com/Kludex/python-multipart) from 0.0.18 to 0.0.20.
- [Release notes](https://github.com/Kludex/python-multipart/releases)
- [Changelog](https://github.com/Kludex/python-multipart/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Kludex/python-multipart/compare/0.0.18...0.0.20)

---
updated-dependencies:
- dependency-name: python-multipart
  dependency-version: 0.0.20
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump einops from 0.8.0 to 0.8.1 in /backend

Bumps [einops](https://github.com/arogozhnikov/einops) from 0.8.0 to 0.8.1.
- [Release notes](https://github.com/arogozhnikov/einops/releases)
- [Commits](https://github.com/arogozhnikov/einops/compare/v0.8.0...v0.8.1)

---
updated-dependencies:
- dependency-name: einops
  dependency-version: 0.8.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump pypandoc from 1.13 to 1.15 in /backend

Bumps [pypandoc](https://github.com/JessicaTegner/pypandoc) from 1.13 to 1.15.
- [Release notes](https://github.com/JessicaTegner/pypandoc/releases)
- [Changelog](https://github.com/JessicaTegner/pypandoc/blob/master/release.md)
- [Commits](https://github.com/JessicaTegner/pypandoc/compare/v1.13...v1.15)

---
updated-dependencies:
- dependency-name: pypandoc
  dependency-version: '1.15'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

tjbck committed:
refac: $user

weisser-dev committed:
[FEAT]-Adjust Translations for temporary chat

lukasz-pekala committed:
fix: update Polish translations for clarity and accuracy

weisser-dev committed:
[improvement] default permission for new groups is false for enforce temp chat

lukasz-pekala committed:
fix: improve Polish translations for clarity and accuracy

tjbck committed:
Merge pull request #12260 from weisser-dev/feat-default-value-for-enforce-temp-chat

[improvement] default permission for new groups is false for enforce …

tjbck committed:
Merge pull request #12253 from open-webui/dependabot/pip/backend/dev/python-multipart-0.0.20

build(deps): bump python-multipart from 0.0.18 to 0.0.20 in /backend

tjbck committed:
Merge pull request #12255 from open-webui/dependabot/pip/backend/dev/pypandoc-1.15

build(deps): bump pypandoc from 1.13 to 1.15 in /backend

tjbck committed:
Merge pull request #12254 from open-webui/dependabot/pip/backend/dev/einops-0.8.1

build(deps): bump einops from 0.8.0 to 0.8.1 in /backend

lukasz-pekala committed:
fix: update Polish translations to sound more natural

weisser-dev committed:
upated like in PR discussed

weisser-dev committed:
Merge branch 'open-webui:dev' into dev

lukasz-pekala committed:
fix: update Polish translation for "Capture" to improve accuracy

tjbck committed:
Merge pull request #12262 from lukasz-pekala/fb/fixes-to-polish-translation

i18n: fixes to polish translation

tjbck committed:
Merge pull request #12259 from weisser-dev/dev

[FEAT] adjust Translations for temporary chat

panda44312 committed:
i18n: zh-cn

panda44312 committed:
fix

TiancongLx committed:
i18n: update zh-TW

tjbck committed:
Merge pull request #12269 from panda44312/patch-9

i18n: zh-cn

tjbck committed:
Merge pull request #12270 from TiancongLx/dev

i18n: update zh-TW

Saidoua committed:
Arabic Translation

Xelaph committed:
Update translation.json

Xelaph committed:
Update translation.json

Xelaph committed:
Update translation.json

Xelaph committed:
Update translation.json

paddy313 committed:
refactor: move MistralLoader to a separate module and just use the requests package instead of mistralai

lowlyocean committed:
Pin onnxruntime to 1.20.1 to address SIGILL on certain arm64 hosts

paddy313 committed:
refactor: update MistralLoader documentation and adjust parameters for signed URL retrieval

OriginalSimon committed:
Update translation.json

landerrosette committed:
refactor: standardize filter valve retrieval logic

tjbck committed:
Merge pull request #12299 from lowlyocean/rpi_onnxruntime

fix: Allow v0.6.0 to run on Raspberry Pi

tjbck committed:
Merge pull request #12278 from Saidoua/main

**i18n** Standard Arabic Translation

tjbck committed:
Merge pull request #12294 from Xelaph/dev

**i18n**: Update Dutch translations

tjbck committed:
Merge pull request #12304 from OriginalSimon/main

18n: Update Ukrainian translation

panda44312 committed:
fix

tjbck committed:
Merge pull request #12310 from landerrosette/fix_filter_priority

fix: Filter function’s priority valve has no effect

paddy313 committed:
refactor: update import path for MistralLoader

SadmL committed:
[i18n] Russian localization update

Fixed typos and translated some lines

silentoplayz committed:
fix

my dev environment works again!

silentoplayz committed:
Update Chats.svelte

Added Archived Chats option in Chats settings section

tjbck committed:
Merge pull request #12331 from panda44312/patch-10

i18n: zh-cn fix

tjbck committed:
Merge pull request #12307 from paddy313/feature/mistral_ocr

feat: Added support for Mistral OCR for Content Extraction

tjbck committed:
Merge pull request #12353 from SadmL/dev

[i18n] Russian localization update

tjbck committed:
Merge pull request #12364 from silentoplayz/archive-chats-option

feat: Added Archived Chats option in Chats settings section

CityOfBunbury committed:
Added OAUTH_USE_PICTURE_CLAIM env var

Added OAUTH_USE_PICTURE_CLAIM to config.py

Added check to oauth.py on OAUTH_USE_PICTURE_CLAIM, to decide whether to user the profile picture in the claim or the default user.png

silentoplayz committed:
Update env.py

Revert

tjbck committed:
Merge pull request #12355 from silentoplayz/logging-fix

fix: update datetime and logging imports

tjbck committed:
fix: admin folder deletion issue

tjbck committed:
enh: unload hybrid model if set to False

tjbck committed:
feat: direct tools user permissions

tjbck committed:
refac: settings ui styling

tjbck committed:
Merge pull request #12376 from MushroomLamp-COB/main

feat: Add option to disable OAuth profile pictures

tjbck committed:
refac

tjbck committed:
refac: rm OAUTH_USE_PICTURE_CLAIM

tjbck committed:
refac: use selected model for merge response

tjbck committed:
fix: tool server api key not being sent

tjbck committed:
refac: remove `None` params

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac: styling

tjbck committed:
enh: image tool response

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac: styling

tjbck committed:
refac: styling

tjbck committed:
refac

tjbck committed:
fix: tls cert requirement

tjbck committed:
refac: message edit

silentoplayz committed:
fix: don't show export button if nothing to export

silentoplayz committed:
Update Feedbacks.svelte

fix

gaby committed:
Add query param to remove content from GET /api/v1/files

silentoplayz committed:
fix: funcs, models, prompts, & tools conditional export button

gaby committed:
Fix API_KEY_ALLOWED_ENDPOINTS

thirdpoler committed:
i18n: add Tibetan language

Commit:
make content parameter optional in OpenAI chat completion API endpoint

gaby committed:
Rename field to include_content

mahenning committed:
Added missing parameter for query_doc_with_hybrid_search.

tjbck committed:
Merge pull request #12447 from floriankick/fix-openai-api-empty-message-after-tool-call

fix: make content parameter optional in OpenAI chat completion API endpoint

tjbck committed:
Merge pull request #12450 from mahenning/fix-missing-parameter-rag

Fix: Add missing parameter for query_doc_with_hybrid_search

tjbck committed:
enh: openapi tool server custom path

tjbck committed:
refac: input prompt

tjbck committed:
refac

tjbck committed:
Merge pull request #12439 from narutopden/i18n-add-Tibetan-translation

i18n: add Tibetan language

tjbck committed:
Merge pull request #12433 from gaby/fix-allowed-endpoints

fix: API Key Endpoint Restrictions for Dynamic Paths

tjbck committed:
refac: tools styling

tjbck committed:
refac: styling

tjbck committed:
chore: format

tjbck committed:
fix: temp chat

tjbck committed:
Merge pull request #12426 from silentoplayz/fix-feedback

fix: Conditionally hide export buttons

Phlogi committed:
Update utils.py

Avoid running any tasks for collections that failed to fetch data (have assigned None)

tjbck committed:
Merge pull request #12476 from Phlogi/dev-hybrid-search

fix: Avoid secondary error when collection is empty in parallel hybrid search

Ithanil committed:
tweak default rag template to be more coherent and improve consistency with citations

theredcat committed:
i18n: improve fr translation

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
Merge pull request #12477 from Ithanil/improved_rag_template

fix: more coherent default RAG template

tjbck committed:
Merge pull request #12478 from theredcat/main

Improve FR translation

tjbck committed:
refac: tool servers

tjbck committed:
refac

tjbck committed:
refac: tools removed UNNECESSARY CODE

tjbck committed:
refac

tjbck committed:
refac: substandard codebase overhauled

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

dan-sullivan committed:
Fallback from desc to summary to placeholder

Kylapaallikko committed:
Update translation.json

gaby committed:
Make auth error messages generic

gaby committed:
Fix formatting issues

tjbck committed:
refac

tjbck committed:
chore: format

FabioPolito24 committed:
fix: fix for text file handling with docling

tjbck committed:
Merge branch 'dev' into dev

tjbck committed:
Merge pull request #12480 from Kylapaallikko/dev

i18n: Update Finnish Translation

tjbck committed:
Merge pull request #12481 from gaby/generic-errors

fix: Improve auth error messages

tjbck committed:
Merge pull request #12486 from FabioPolito24/text-file-handling-docling

fix: text file handling with docling

tjbck committed:
doc: changelog

tjbck committed:
chore: bump

tjbck committed:
Merge pull request #12373 from open-webui/dev

0.6.1

tjbck committed:
refac

tjbck committed:
Merge pull request #12491 from open-webui/dev

refac

TiancongLx committed:
i18n: update zh-TW

tjbck committed:
Merge pull request #12496 from dan-sullivan/fix-12479/add-desc-to-openapi-tools

fix: openapi tools should pass description to model not summary

tjbck committed:
refac

TiancongLx committed:
Merge remote-tracking branch 'upstream/dev' into dev

* upstream/dev:
  refac
  Fallback from desc to summary to placeholder

Ithanil committed:
fix web results all getting the same source id when bypassing embedding and retrieval

Ithanil committed:
fix web results all getting the same source id when using embedding and retrieval

newnol committed:
Update translation.json

panda44312 committed:
i18n: zh-cn

panda44312 committed:
i18n: fix

gaby committed:
Merge branch 'open-webui:main' into fix-12237

gaby committed:
Fix formatting

gaby committed:
Fix dependabot configuration

Ithanil committed:
only keep URLs as sources for which the content could actually be retrieved

gaby committed:
Update index.ts

tjbck committed:
Merge pull request #12515 from gaby/fix-dependabot

chore: Fix dependabot configuration

tjbck committed:
Merge pull request #12517 from Ithanil/only_keep_retrieved_urls

fix: only keep URLs as sources for which the content was actually retrieved

tjbck committed:
Merge pull request #12511 from panda44312/patch-11

i18n: zh-cn

tjbck committed:
Merge pull request #12493 from TiancongLx/dev

i18n: update zh-TW

gaby committed:
Bump Python base Docker image to 3.12

tjbck committed:
Merge pull request #12510 from newnol/patch-1

i18n language pack vietnamese

tjbck committed:
Merge pull request #12431 from gaby/fix-12237

feat: Allow making content optional when listing all files

tjbck committed:
refac

gaby committed:
Optimize GitHub actions

tjbck committed:
Merge pull request #12520 from gaby/bump-docker

chore: Bump Python Docker image to python3.12

tjbck committed:
Merge pull request #12506 from Ithanil/fix_web_result_source_ids

fix: fix web results all getting the same source id when bypassing embedding and retrieval

tjbck committed:
refac

tjbck committed:
Merge pull request #12521 from gaby/optimize-workflow

chore: Only run GitHub workflows when needed

tjbck committed:
refac

tjbck committed:
chore: format

tjbck committed:
Merge pull request #12507 from Ithanil/fix_web_result_collection_source_ids

fix: fix web results all getting the same source id when using embedding and retrieval

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac

tjbck committed:
refac: embedding prefix

tjbck committed:
fix: audio file upload response issue

tjbck committed:
chore: format

tjbck committed:
chore: include `accelerate` dependency

TiancongLx committed:
i18n: update zh-TW

tjbck committed:
Merge pull request #12525 from TiancongLx/dev

i18n: update zh-TW

tjbck committed:
revert

tjbck committed:
refac: citations modal

gaby committed:
Track backend/requirements.txt updates

tjbck committed:
Merge pull request #12526 from gaby/dependabot-update

chore: Dependabot update

tjbck committed:
refac

tjbck committed:
doc: changelog

tjbck committed:
chore: format

tjbck committed:
doc: changelog

tjbck committed:
Merge pull request #12524 from open-webui/dev

0.6.2

nabeelraza-7 committed:
Merge remote-tracking branch 'upstream/main' into searxng-feature
updating with the upstream

nabeelraza-7 committed:
upate: requirements.txt
